### PR TITLE
Run CI tests using GitHub Actions.

### PR DESCRIPTION
**Pull request by antonsviridenko**

### CODE REVIEW

**Comment by antonsviridenko** [comment]
I guess tests will run after this PR is merged, not before.
Doing it for the first time 🙂

**Comment by antonsviridenko** [comment]
ok, librnp needs to be built first...

**Comment by antonsviridenko** [comment]
Looks like the whole content of https://github.com/rnpgp/rnp/tree/master/ci should be reused here, in order to install dependencies & build rnp. I don't like the idea of duplicating this code.

**Comment by antonsviridenko** [comment]
We don't have official binary packages of rnp, right? To use as a build dependency for this extension.

**Comment by antonsviridenko** [comment]
Or maybe we can upload latest successful builds from the master branch of `rnp` to the Packages section of the `rnp` repository and then these builds can be reused here as a build dependency...

**Review by ronaldtse — APPROVED** [review]
Thank you @antonsviridenko !

**Comment by ronaldtse** [comment]
With regards to the build/release issues, @maxirmx is going to help us refactor the whole flow so that we would be able to reuse the build flow as much as possible.

**Comment by maxirmx** [comment]
@ni4 @antonsviridenko 

Test 10

```
$options = array(
		'compression_alg' => 'ZIP',
		'compression_level' => 6,
		'armor' => true,
		'hash' => 'SHA224',
		'creation_time' => 1234337,
		'expiration_time' => 2147483647,
		'file_name' => 'testfilename',
		'file_mtime' => 321337,
		'add_signature' => true,
		'password' => 'password',
		'cipher' => 'CAMELLIA192',
		'aead' => 'EAX',
		'aead_bits' => 0,
		//'flags' => RNP_ENCRYPT_NOWRAP
		);

$encrypted_message = rnp_op_encrypt($ffi, $message, array($key1, $key2), $options);
```
Behavior has changed from v0.16.2 to the latest master.
Pls note the difference at  ```Symmetric-key encrypted session key packet/symmetric algorithm``  below

Is this change expected ?

**On v0.16.2:**
```
    :armored input
    :off 0: packet header 0xc1c04c (tag 1, len 268)
    Public-key encrypted session key packet
        version: 3
        key id: 0x8692588d599ba513
        public key algorithm: 1 (RSA (Encrypt or Sign))
        encrypted material:
            rsa m: 2048 bits
    :off 271: packet header 0xc1c04c (tag 1, len 268)
    Public-key encrypted session key packet
        version: 3
        key id: 0xe72fcdff542c8e16
        public key algorithm: 1 (RSA (Encrypt or Sign))
        encrypted material:
            rsa m: 2045 bits
    :off 542: packet header 0xc346 (tag 3, len 70)
    Symmetric-key encrypted session key packet
        version: 5
        symmetric algorithm: 12 (Camellia-192)
        aead algorithm: 1 (EAX)
        s2k specifier: 3
        s2k hash algorithm: 8 (SHA256)
        s2k salt: 0x6a6e9fa65a049b71
        s2k iterations: 31457280 (encoded as 238)
        aead iv: 0x3a0196b77e7221e3f07647dfa177fa00 (16 bytes)
        encrypted key: 0x26063ad7a9acedf31f85917e15db42df3f6f492bbd07d9a3850682126e13ce7b26d1fa6764373c8c (40 bytes)
    :off 614: packet header 0xd4c1d9 (tag 20, len 665)
    AEAD-encrypted data packet
        version: 1
        symmetric algorithm: 12 (Camellia-192)
        aead algorithm: 1 (EAX)
        chunk size: 0
        initialization vector: 0xa556821cb3f5788cb3328811a04e80e7 (16 bytes)
    armored input
```
**On the master branch:**
```
:armored input
:off 0: packet header 0xc1c04c (tag 1, len 268)
Public-key encrypted session key packet
    version: 3
    key id: 0xa10761bde570fa05
    public key algorithm: 1 (RSA (Encrypt or Sign))
    encrypted material:
        rsa m: 2047 bits
:off 271: packet header 0xc1c04c (tag 1, len 268)
Public-key encrypted session key packet
    version: 3
    key id: 0x9f9b348f5fbf0977
    public key algorithm: 1 (RSA (Encrypt or Sign))
    encrypted material:
        rsa m: 2048 bits
:off 542: packet header 0xc346 (tag 3, len 70)
Symmetric-key encrypted session key packet
    version: 5
    symmetric algorithm: 9 (AES-256)
    aead algorithm: 1 (EAX)
    s2k specifier: 3
    s2k hash algorithm: 8 (SHA256)
    s2k salt: 0x827636b074ad0150
    s2k iterations: 30408704 (encoded as 237)
    aead iv: 0x8e185c39128c04094ebb8ff0d14fe703 (16 bytes)
    encrypted key: 0x5284c46cd30a3ddb6166345a3659d10cb88d36e767307c4da021c05cad41d9ba50b85fe730a47aa2 (40 bytes)
:off 614: packet header 0xd4c1db (tag 20, len 667)
AEAD-encrypted data packet
    version: 1
    symmetric algorithm: 12 (Camellia-192)
    aead algorithm: 1 (EAX)
    chunk size: 0
    initialization vector: 0xa5610153b95cca3d234dfd93858bfb9a (16 bytes)
armored input
    ```

**Comment by ni4** [comment]
@maxirmx That's because of this PR and linked to it issue: https://github.com/rnpgp/rnp/pull/1995
PHP code should be updated to specify symmetric algorithm while adding SKESK via the `rnp_op_encrypt_add_password()` call.

**Review by ronaldtse — APPROVED** [review]

**Comment by antonsviridenko** [comment]
@maxirmx 
By adding the new option "kek_cipher" you are updating the PHP extension API,
therefore this update must be accompanied by updated API version and you also
should update the extension docs like here
https://github.com/antonsviridenko/doc-en/tree/rnp
(i.e. clone the latest "doc-en" repo from PHP project, add description for "kek_cipher" in `doc-en/tree/rnp/reference/rnp/functions/rnp-op-encrypt.xml`, make a pull request to upstream)

I'd propose to avoid adding this new option, 
the option "cipher" can be reused instead, i.e. "cipher" value can be passed to both `rnp_op_encrypt_add_password()` and `rnp_op_encrypt_set_cipher()`.
In order to make the tests pass on both v16.2 and latest master version, 
test 10 can be updated to check for presence of at least one substring "12 (Camellia-192)".
Now it expects exactly two of them, 
but for example  the function  `rnptest_checkdump()` can be updated to accept an optional third parameter that changes
semantics  of counting substrings"

File: package.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
  </lead>
  <date>2022-10-25</date>
  <version>
-  <release>0.1.1</release>
+  <release>0.2.0</release>
   <api>0.1</api>
```
Comment (antonsviridenko) [inline package.xml:22]: If you decide to add new option "kek_cipher", <api>0.1</api> should be incremented too.

File: package.xml [hunk 2]
```diff
@@ -78,7 +78,7 @@
  <changelog>
   <release>
    <version>
-    <release>0.1.1</release>
+    <release>0.2.0</release>
```
Comment (antonsviridenko) [inline package.xml:81]: In a changelog section  new entries should be appended instead of replacing the old version.

File: tests/010.phpt [hunk 3]
```diff
@@ -65,6 +65,7 @@ $options = array(
 		'add_signature' => true,
 		'password' => 'password',
 		'cipher' => 'CAMELLIA192',
+		'kek_cipher' => 'CAMELLIA192',
```
Comment (antonsviridenko) [inline tests/010.phpt:68]: I propose to reuse the value of 'cipher' instead of adding new 'kek_cipher'

**Comment by antonsviridenko** [comment]
https://pear.php.net/manual/en/developers.packagedef.intro.php

**Comment by maxirmx** [comment]
> https://pear.php.net/manual/en/developers.packagedef.intro.php

@antonsviridenko, I am not php expert. Could you please explain why 6 months version 1 was good but  now it is the time to update it ?
Thank you

**Comment by antonsviridenko** [comment]
hmm, looks like I posted the wrong link, should be https://pear.php.net/manual/en/guide.developers.package2.intro.php
Our package.xml version is already 2.0

I can implement & commit my  changes proposed here https://github.com/rnpgp/php-rnp/pull/9#issuecomment-1495959422
myself if you want.

**Comment by ronaldtse** [comment]
@antonsviridenko please feel free to make updates and merge this when ready. Thanks!

**Comment by ni4** [comment]
> For me ot does not match the idea behind [rnpgp/rnp#1995](https://github.com/rnpgp/rnp/pull/1995) , but ok, let's enforce the same cipher @antonsviridenko pls have a look at the updated version

FFI API was designed to 'allow user whatever he wants and OpenPGP allows', including having different CEK/KEK ciphers. However, PHP wrappers should make things simpler. Btw, GnuPG issuess a warning if CEK/KEK ciphers differ.

**Comment by maxirmx** [comment]
https://github.com/rnpgp/php-rnp/issues/7   ?

**Comment by antonsviridenko** [comment]
@maxirmx let's merge this one first

**Review by ni4 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**